ocpp-v1_6/src/main/java/eu/chargetime/ocpp/JSONCommunicator.java

```diff
@@ -154,7 +154,10 @@ protected Message parse(Object json) {
             ((CallErrorMessage) message).setErrorCode(array.get(INDEX_CALLERROR_ERRORCODE).getAsString());
             ((CallErrorMessage) message).setErrorDescription(array.get(INDEX_CALLERROR_DESCRIPTION).getAsString());
             ((CallErrorMessage) message).setRawPayload(array.get(INDEX_CALLERROR_PAYLOAD).toString());
+        } else {
+            throw new IllegalArgumentException("Unknown message type");
```

Comment:
I'd like to suggest you to add JSON body to the log.

Reply:
Done